### PR TITLE
[#LW-1903] Loocho-IntegrationICloud : Enable the 2 Factor Authentications

### PR DESCRIPTION
**Pull request by parany**

### CODE REVIEW

**Comment by parany** [comment]
I created a new module "Auth2" and left the version 1 intact. This second version is written in generator-style, so it's easy to read and follow,  and  also supports 2-factor authentication.

File: modules/auth2.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    }
+
+  }
+
+  /**
+   * Log out current user with current dsid
+   */
+  *_logout() {
+    try {
+      yield this.req.get({
```
Comment (zo-tahiana) [inline modules/auth2.js:115]: Maybe `yield` is optional here

**Review by zo-tahiana — APPROVED** [review]
I left one comment and it's better to add YUIDoc for each function :p

File: modules/auth2.js (hunk 1 above)
Comment (parany) [reply modules/auth2.js:115]: We need to await for this request to be finished to be able to catch the error.

Comment (zo-tahiana) [reply modules/auth2.js:115]: Understood

Comment (parany) [reply modules/auth2.js:115]: I added a description to all methods.

**Comment by parany** [comment]
@vincentjacques Please review this PR.

**Review by vincentjacques — APPROVED** [review]

**Pull request merged**